File: backend/Routes/medicine.js
const express = require('express');
const router = express.Router();
const multer = require('../Middlewares/multer')
const fetchuser = require('../Middlewares/fetchuser');
const MedicineSchema = require('../Schemas/MedicineSchema');
const SeniorCitizensSchema = require('../Schemas/SeniorCitizensSchema');

router.post('/add', fetchuser, multer.upload.single('image'), async (req, res) => {
    try {
        const userFind = await SeniorCitizensSchema.findOne({ uniqueId: req.user.uniqueId })
        const newMedicine = new MedicineSchema({
            seniorId: userFind.id,
            name: req.body.name,
            days: req.body.days,
            time: req.body.time,
            duration: parseInt(req.body.duration),
            image: `${process.env.URI}api/image/${req.file.filename}`
        })

        const saved = await newMedicine.save()
        console.log(saved)
        res.status(200).json(saved)
    } catch (err) {
        console.log(err.message)
        res.status(500).json(err.message);
    }
})

router.get('/get', fetchuser, async (req, res) => {
    try {
        // console.log(req.user)
        const userFind = await SeniorCitizensSchema.findOne({ uniqueId: req.user.uniqueId })
        // console.log(userFind)
        const findMedicines = await MedicineSchema.find({ seniorId: userFind.id })
        res.status(200).json(findMedicines)
    } catch (err) {
        console.log(err.message)
        res.status(500).json(err.message);
    }
})

module.exports = router

File: surakshak/lib/models/medicine.dart
class MedModel {
  String? seniorId;
  String? name;
  String? days;
  String? time;
  int? duration;
  String? image;
  String? sId;
  int? iV;

  MedModel(
      {this.seniorId,
      this.name,
      this.days,
      this.time,
      this.duration,
      this.image,
      this.sId,
      this.iV});

  MedModel.fromJson(Map<String, dynamic> json) {
    seniorId = json['seniorId'];
    name = json['name'];
    days = json['days'];
    time = json['time'];
    duration = json['duration'];
    image = json['image'];
    sId = json['_id'];
    iV = json['__v'];
  }
}
